uploads: Ask user about deleting uploaded files when removed from edited message

wandrew0 opened the pull request:
Complete the unfinished work from https://github.com/zulip/zulip/pull/25764 in May 2023. When a user edits a message, attachments can be removed or changed. When the user hits 'save', if any attachments are removed, a modal will pop up asking the user to delete them.

   -  merge #25764, resolve conflict, fix bugs.
   -  add test cases.

Fixes: https://github.com/zulip/zulip/issues/25525

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**


![Screenshot from 2024-05-03 09-38-15](https://github.com/zulip/zulip/assets/73965466/d6b6b7d3-4781-4892-9e04-44bc6e1f335e)
![Screenshot from 2024-05-03 09-38-34](https://github.com/zulip/zulip/assets/73965466/a6d3559f-e60c-4912-8f92-265ba758eb54)

<details>



<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ x] Each commit is a coherent idea.
- [ x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ x] Visual appearance of the changes.
- [x ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ x] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

wandrew0 commented:
I have fixed the commit comments, and the PR is ready for review. Thanks!

alya commented:
The "Manage uploaded files" link feels out of place to me. Let's try moving it below the list of files, and see if that helps.

wandrew0 commented:
@alya I have removed the 'edit history' link and added an error state screenshot and screenshots with 'manage uploaded files' at the bottom. (see updated PR). Let me know which option you like more. Thanks.

alya commented:
Thanks! I reopened the PR and posted a question [on CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/delete.20attachments.20from.20mesage.20history) for broader feedback.

alya commented:
Is there a reason you've been closing your PRs?

wandrew0 commented:
> Is there a reason you've been closing and reopening your PRs?
@alya This PR frequently gets merge conflict because it changes a volatile file changelog.md. It will create an additional commit if I opt to "resolve" the merge conflict. As  I don't want to pollute my commit history with the 'merge/resolve conflict' commits, I stash away my commit changes, reset the branch to Zulip main, and reapply my changes so my commits can be clean. The side effect is that Github automatically closes PR when I reset my branch to main. Therefore, I have to reopen it. 
I'm still learning Git, and I apologize for the multiple close/open actions, which can be confusing for a reviewer. I will research Git to handle situations like this better. Thank you.

alya commented:
Yeah, that sounds like a more complicated workflow than you need. In any case, you can always use `rebase -i` to squash any extra commits.

alya commented:
https://zulip.readthedocs.io/en/latest/git/index.html might help, or asking in #git-help on CZO.

wandrew0 commented:
@alya I have made the code changes and uploaded the latest screenshots. It's ready for review.

alya commented:
Thanks, the screenshots look good to me! @timabbott do you wan to take a look next?

Joelute commented:
Hey @alya, this PR could be marked as close as it has been completed in #31161.

laurynmm commented:
Thanks for that ping @Joelute. And @wandrew0, thanks for working on this issue!
